### PR TITLE
Problem parsing unknown Javadoc tag

### LINKED ISSUE #3750: Problem parsing unknown Javadoc tag
When Javadoc contains unknown tags as `{@return}` in the following test, the parser ends up with a parse-to-print failure:
```java
    @Test
    void javadocUnknownTag() {
        rewriteRun(
          java(
            """
              interface Test {
              	/**
              	 * {@return 42}
              	 */
              	int foo();
              }
              """
          )
        );
    }
```

### PR DESCRIPTION
**Pull request by TheMarvelFan**
Referencing issue 
- fixes #3750

## What's changed?

Modified the sourceBefore() method and visiteReturn() method to handle leading curly brace '{' with return annotations, i.e., cases like '{@return}' and '{@return}'. I will be adding commits that also modify the JavaDocPrinter to include the leading '{' when printing the modified javadoc.

## Anyone you would like to review specifically?

@traceyyoshima or @timtebeek 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Thanks for getting this started @TheMarvelFan ! I've removed two classes from your other PR, which squash merge should resolve
- https://github.com/openrewrite/rewrite/pull/3784

As for the changes here probably best to add a test to verify that the changes here correctly handle the cases mentioned in https://github.com/openrewrite/rewrite/issues/3750#issuecomment-1854825540

**Comment by timtebeek** [comment]
Then once we're satisfied with the changes (but only then!), we should replicate the same to the other parsers to also fix this in Java 8, 11 and 21.

**Comment by TheMarvelFan** [comment]
Hi @timtebeek,
Please review the modified files, and suggest changes if required.

**Comment by timtebeek** [comment]
Hi @TheMarvelFan ; you seem to have added two files to the root of your repository fork:
https://github.com/TheMarvelFan/rewrite/tree/rewrite1
image

You likely intended to add those elsewhere; what's the reason you're using the file upload and not a git push?

**Comment by TheMarvelFan** [comment]
Hi,
I initially intended to contribute to both this issue and issue #3757 using the same repository fork. For this purpose, I created another branch named "rewrite1", but I think I made a mistake somwhere, and when I tried pushing to rewrite one from origin on my local machine, I got an error which told me to "fetch first" and then push. I tried to fetch as well, but the error remained. So, I decided to upload the files to the rewrite1 branch instead of pushing. I am still not sure if the same cloned repository can be used to push different commits to two different open pull requests of the same upstream. I apologize for the inconvenience due to this.

**Comment by timtebeek** [comment]
Yes you can use the same fork for multiple open pull requests, but it helps to start a branch for each new development you start, and not commit to the main branch. Otherwise you indeed run the risk of conflicts and diverging commits on the main branch.

**Comment by TheMarvelFan** [comment]
I understand. I am still quite new to contributing to open source, but with time I will get better. Is there something I can do right now to correct this problem?

**Review by traceyyoshima — COMMENTED** [review]
Hi @TheMarvelFan! The changes are looking good so far, thank you for fixing this bug!

I've added a few comments with information on the changes. Please let me know if you have any questions!

**Comment by TheMarvelFan** [comment]
Hi @traceyyoshima,
Thanks for suggesting the changes. Please review the modified files and let me know if more changes are required.

File: rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java [hunk 1]
```diff
@@ -1197,4 +1204,17 @@ public J visitParameterizedType(ParameterizedTypeTree node, Space fmt) {
             return new J.ParameterizedType(randomId(), fmt, Markers.EMPTY, id, JContainer.build(expressions), typeMapping.type(node));
         }
     }
+
+    private static class LeadingBraceMarker implements Marker {
```
[the remark this answers is not in the document]
Comment (timtebeek) [reply rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java:1211]:
```suggestion
    @Value
    @With
    private static class LeadingBraceMarker implements Marker {
```

File: rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java [hunk 2]
```diff
@@ -1197,4 +1207,19 @@ public J visitParameterizedType(ParameterizedTypeTree node, Space fmt) {
...
+        @Override
+        public UUID getId() {
+            return uuid;
+        }
+
+        @Override
+        public <M extends Marker> M withId(UUID id) {
+            return null;
+        }
+        // You can add any additional fields or methods if needed
```
Comment (timtebeek) [inline rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java:1223]:
```suggestion
```

**Comment by traceyyoshima** [comment]
> Once we're satisfied with the changes (but only then!), we should replicate the same to the other parsers to also fix this in Java 8, 11 and 21.

I think the changes look good and we're ready to update the other parsers. What do you think?

File: rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java [hunk 3]
```diff
@@ -753,8 +756,16 @@ private JavaType.Variable fieldReferenceType(DCTree.DCReference ref, @Nullable J
 
     @Override
     public Tree visitReturn(ReturnTree node, List<Javadoc> body) {
-        body.addAll(sourceBefore("@return"));
-        return new Javadoc.Return(randomId(), Markers.EMPTY, convertMultiline(node.getDescription()));
+        List<Javadoc> before;
+        Markers markers = Markers.EMPTY;
+        if (source.startsWith("{", cursor)) {
+            markers = markers.addIfAbsent(Markers.build(Collections.singletonList(new LeadingBraceMarker(UUID.randomUUID()))));
```
Comment (traceyyoshima) [inline rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java:762]: A few changes are necessary here:

- `addIfAbsent` adds a new Marker if it doesn't exist. So, the argument expects a `Marker`.
- Replace `UUID.randomUUID()` with `Tree.randomId()`

```suggestion
markers = markers.addIfAbsent(new LeadingBrace(Tree.randomId()));
```

File: rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java [hunk 4]
```diff
@@ -1197,4 +1209,10 @@ public J visitParameterizedType(ParameterizedTypeTree node, Space fmt) {
             return new J.ParameterizedType(randomId(), fmt, Markers.EMPTY, id, JContainer.build(expressions), typeMapping.type(node));
         }
     }
+
+    @Value
+    @With
+    private static class LeadingBraceMarker implements Marker {
```
Comment (traceyyoshima) [inline rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java:1215]:
```suggestion
    @Value
    @With
    private static class LeadingBrace implements Marker {
        UUID id;
    }
```

**Review by traceyyoshima — COMMENTED** [review]
Nearly there!

File: rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java [hunk 5]
```diff
@@ -1197,4 +1209,10 @@ public J visitParameterizedType(ParameterizedTypeTree node, Space fmt) {
             return new J.ParameterizedType(randomId(), fmt, Markers.EMPTY, id, JContainer.build(expressions), typeMapping.type(node));
         }
     }
+
+    @Value
+    @With
+    private static class LeadingBrace implements Marker {
+        private UUID uuid;
```
Comment (traceyyoshima) [inline rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17JavadocVisitor.java:1216]:
```suggestion
        UUID id;
```

**Comment by traceyyoshima** [comment]
@TheMarvelFan I've updated the PR with a few finishing touches, thank you for your work on this PR!

**Review by timtebeek — APPROVED** [review]

**Pull request merged**